### DISCUSSION TITLE
getUploadedFileUrlUsing method gone in v3

### DISCUSSION THREAD

**Discussion by bomshteyn** [filamentphp/filament#7432] [Help]
I have been using the `getUploadedFileUrlUsing` method on `FileUpload` fields to enable download of files from private folder using a custom URL route. 

Example:
```PHP
->getUploadedFileUrlUsing(function ($file, callable $set) {
    return $set('file_url', route('illustration.file.show', $file));
})
```

I see this method is gone now from V3. What is an equivalent method i can use?

I saw the new BaseFileUpload contains `getUploadedFileUsing` method and tied that but it didn't work.

**Comment by danharrin** [comment]
`getUploadedFileUsing()` is the replacement but it returns a different data structure. Be aware this is an internal method I think
